### PR TITLE
Updated reducer.sh script to work with MariaDB Galera Cluster

### PR DESCRIPTION
**Pull request by mariadb-RameshSivaraman**

### CODE REVIEW

File: reducer.sh [hunk 1]
```diff
@@ -2064,7 +2056,10 @@ start_mysqld_or_valgrind_or_pxc(){
   STARTUPCOUNT=$[$STARTUPCOUNT+1]
 }
 
-start_pxc_main(){
+start_mdg_main(){
+  #clean existing processes
+  ps -ef | grep -e  'n1.cnf\|n2.cnf\|n3.cnf' | grep -v grep |  grep $EPOCH | awk '{print $2}' | xargs kill -9 >/dev/null 2>&1 || true
```
Comment (mariadb-RoelVandePaar) [inline reducer.sh]: Maybe use `grep -E` and change `\|` to `|` to be uniform with other code, i.e.
```
$ echo -ne '1\n2\n3\n'
1
2
3
$ echo -ne '1\n2\n3\n' | grep -e '2|3'
$ echo -ne '1\n2\n3\n' | grep -e '2\|3'
2
3
$ echo -ne '1\n2\n3\n' | grep -E '2|3'
2
3
```

Comment (mariadb-RoelVandePaar) [reply reducer.sh]: Please do the same with other existing occurences of
```
grep -e  'node1_socket\|node2_socket\|node3_socket'
```
To
```
grep -E 'node1_socket|node2_socket|node3_socket'
```

File: reducer.sh [hunk 2]
```diff
@@ -3176,11 +3160,11 @@ process_outcome(){
...
 }
 
-stop_mysqld_or_pxc(){
+stop_mysqld_or_mdg(){
   SHUTDOWN_TIME_START=$(date +'%s')
   MODE0_MIN_SHUTDOWN_TIME=$[ $TIMEOUT_CHECK + 10 ]
-  if [[ $USE_PXC -eq 1 || $USE_GRP_RPL -eq 1 ]]; then
-    (ps -ef | grep -e  'node1_socket\|node2_socket\|node3_socket' | grep -v grep |  grep $EPOCH | awk '{print $2}' | xargs kill -9 >/dev/null 2>&1 || true)
+  if [[ $USE_MDG -eq 1 || $USE_GRP_RPL -eq 1 ]]; then
+    (ps -ef | grep -e  'n1.cnf\|n2.cnf\|n3.cnf' | grep -v grep |  grep $EPOCH | awk '{print $2}' | xargs kill -9 >/dev/null 2>&1 || true)
```
Comment (mariadb-RoelVandePaar) [inline reducer.sh]: Idem

File: reducer.sh [hunk 3]
```diff
@@ -580,8 +580,8 @@ abort(){  # Additionally/also used for when echo_out cannot locate $INPUTFILE an
     echo_out "[Abort] Best testcase thus far: $INPUTFILE (= input file; no optimizations were successful)"
   fi
   echo_out "[Abort] End of dump stack"
-  if [ $USE_PXC -eq 1 ]; then
-    echo_out "[Abort] Ensuring any remaining PXC nodes are terminated and removed"
+  if [ $USE_MDG -eq 1 ]; then
+    echo_out "[Abort] Ensuring any remaining MDG nodes are terminated and removed"
     (ps -ef | grep -e  'node1_socket\|node2_socket\|node3_socket' | grep -v grep |  grep $EPOCH | awk '{print $2}' | xargs kill -9 >/dev/null 2>&1 || true)
```
Comment (mariadb-RoelVandePaar) [inline reducer.sh]: Does this need changing to `n1.cnf` etc. ?

Comment (mariadb-RoelVandePaar) [reply reducer.sh]: (And any other alike statements?)

File: new_text_string.sh [hunk 4]
```diff
@@ -39,7 +39,7 @@ if [ -z "${MYSQLD}" ]; then
   fi
 fi
 
-if [[ ${MDG} -eq 1 ]]; then
+if [[ ${MDG} -eq 1 || ${USE_MDG} -eq 1 ]]; then
```
Comment (mariadb-RoelVandePaar) [inline new_text_string.sh]: Can `USE_MDG` and `MDG` be consolidated?

Comment (mariadb-RoelVandePaar) [reply new_text_string.sh]: Discussed; 
- Consolidate to `MDG`
- Add comment   `# MDG variable is used for pquery-run.sh, pquery-results.sh, reducer.sh etc.`

**Review by mariadb-RoelVandePaar — APPROVED** [review]
Discussed, looks good

File: reducer.sh (hunk 1 above)
Comment (mariadb-RoelVandePaar) [reply reducer.sh]: Discussed further on practicality, an agreed
```
ps -ef | grep -E 'n1.cnf|n2.cnf|n3.cnf' | grep $EPOCH | awk '{print $2}' | xargs -I{} kill -9 {} >/dev/null 2>&1 || true
```

**Review by mariadb-RoelVandePaar — APPROVED** [review]

File: new_text_string.sh (hunk 4 above)
Comment (mariadb-RameshSivaraman) [reply new_text_string.sh]: done

**Pull request merged**